### ISSUE TITLE
Implement queue portability

### ISSUE THREAD

**Issue by bradgrantham-lunarg** [LunarG/gfxreconstruct#856]
From #192 

Differences with the number of available queue families and number of queues available per family could potentially be handled on replay with some sort of mapping, but care would need to be taken when consolidating queue submissions from N queues to 1 queue to avoid synchronization issues and ensure submissions are merged in the correct order. It may be simpler to handle this case by using the dev-sim layer to report fewer available queues on capture.

**Comment by andrew-lunarg** [comment]
There's been one-queue-for-everything implementations around for a while and I think even implementations offering tens of queues still have at least one queue that can do everything. It would be worth going through all major implementations and seeing if there was a need to provide some limited "virtual queue" support to bridge any corner cases that using Vulkan Profiles wouldn't cover.

**Comment by per-mathisen-arm** [comment]
I am a bit worried that this would change the traced workload a lot in terms of both (different) functionality and (worse) performance, to the point where what you've traced isn't really representative of what would have run on a device, though. Not sure what the best solution is here, but presenting only a single queue seems a step too far. (Also I suspect we will see more specialized queues going forward, eg for video, that won't fit well with the single queue view.)

**Comment by andrew-lunarg** [comment]
Agreed. Queue remapping / virtualisation / limiting features would always be optional and to service those who wanted to generate traces that could be replayed across many Vulkan implementations in a portable manner. Users who want that, would like large roughly game-content representative traces that exercises lots of API with less maintenance burden than making multiple versions of a trace per platform/device/driver revision. There would always be the capability, probably by default, to get the rawest non-portable capture possible.

**Comment by andrew-lunarg** [comment]
> (From https://gitlab.khronos.org/vulkan/Vulkan-SDK-Packaging/-/issues/1359)
> 
> Using a capture from nvidia on intel, in this case there are no fatal errors and -m rebind fixes all but 2 VUIDs, these are still reported:
> ...

So we don't lose the context, this is `vkcube`, not a triple-A game trying to use all queues optimally for async resource uploads, compute, drawing, sparse etc.
